Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/IISWebAppMgmtV2/AppCmdOnTargetMachines.ps1

```diff
@@ -0,0 +1,925 @@
+
+    $appCmdArgs = [string]::Format('set config "{0}" /section:anonymousAuthentication /enabled:{1} /commit:apphost', $websiteName, $anonymousAuthentication)
+    $command = "`"$appCmdPath`" $appCmdArgs"
+    Write-Verbose "Setting anonymous authentication for website '$websiteName' to $anonymousAuthentication."
+    Write-Verbose "Running command $command"
+    Run-Command -command $command
+
+    $appCmdArgs = [string]::Format('set config "{0}" /section:basicAuthentication /enabled:{1} /commit:apphost', $websiteName, $basicAuthentication)
+    $command = "`"$appCmdPath`" $appCmdArgs"
+    Write-Verbose "Setting basic authentication for website '$websiteName' to $basicAuthentication."
+    Write-Verbose "Running command $command"
+    Run-Command -command $command
+
+    $appCmdArgs = [string]::Format('set config "{0}" /section:windowsAuthentication /enabled:{1} /commit:apphost', $websiteName, $windowsAuthentication)
+    $command = "`"$appCmdPath`" $appCmdArgs"
+    Write-Verbose "Setting windows authentication for website '$websiteName' to $windowsAuthentication."
+    Write-Verbose "Running command $command"
+    Run-Command -command $command
+}
+
+function Get-PSCredentials {
+
+    param (
+        [string][AllowNull()] $username, 
+        [string][AllowNull()] $password 
+    )
+
+    if ([string]::IsNullOrWhiteSpace($username) -or [string]::IsNullOrWhiteSpace($password)) {
+        return $null
+    }
+
+    $secretPassword = "$password" | ConvertTo-SecureString -AsPlainText -Force
+    $credentials = New-Object System.Management.Automation.PSCredential ("$username", $secretPassword)
+
+    return $credentials
+}
+
+function Invoke-Main
+{
+    param (
+        [string]$ActionIISWebsite,
+        [string]$ActionIISApplicationPool,
+       
+        [string]$CreateApplication,
+        [string]$CreateVirtualDirectory,
+       
+        [string]$VirtualPath,
+        
+        [string]$WebsiteName,
+        [string]$PhysicalPath,
+        [string]$PhysicalPathAuth,
+        [string]$PhysicalPathAuthUserName,
+        [string]$PhysicalPathAuthUserPassword,
+        
+        [string]$AddBinding,
+        [string]$Bindings,
+        
+        [string]$AppPoolName,
+        [string]$DotNetVersion,
+        [string]$PipeLineMode,
+        [string]$AppPoolIdentity,
+        [string]$AppPoolUsername,
+        [string]$AppPoolPassword,
+
+        [string]$ConfigureAuthentication,
+        [string]$AnonymousAuthentication,
+        [string]$BasicAuthentication,
+        [string]$WindowsAuthentication,
+        
+        [string]$AppCmdCommands
+    )
+
+    $physicalPathAuthCredentials = Get-PSCredentials -username $PhysicalPathAuthUserName -password $PhysicalPathAuthUserPassword
+    $appPoolCredentials = Get-PSCredentials -username $AppPoolUsername -password $AppPoolPassword
+    
+    $bindingsArray = ""
+    if (![string]::IsNullOrWhiteSpace($Bindings)) {
+        $bindingsArray = $Bindings | ConvertFrom-Json
```

Comment:
ConvertFrom-Json is available from PS versions 2 and above. Is there any risk ?

Reply:
Don't think so. We should be ok here, especially as this a new major version.